src/bentoml/_internal/service/inference_api.py

```diff
@@ -29,12 +28,15 @@ def __init__(
         user_defined_callback: t.Callable[..., t.Any] | None,
         input_descriptor: IODescriptor[t.Any],
         output_descriptor: IODescriptor[t.Any],
-        name: Optional[str],
-        doc: Optional[str] = None,
-        route: Optional[str] = None,
+        name: str | None = None,
+        doc: str | None = None,
+        route: str | None = None,
     ):
         # Use user_defined_callback function variable if name not specified
-        name = user_defined_callback.__name__ if name is None else name
+        if user_defined_callback is None:
+            name = "" if name is None else name
+        else:
```

Comment:
What's this for?

Reply:
YOu are setting the user_defined_callback in the client to None right. Then the previous logic would fail as None doesn't have a __name__?